qemu/tests/timedrift_with_multi_vms.py

```diff
@@ -0,0 +1,120 @@
+import re
+import logging
+import random
+
+from avocado.utils import process
+
+from virttest import utils_time
+from virttest import env_process
+from virttest import error_context
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Check the timedrift on all the guests when attached all guests to one CPUs
+
+    1) boot the multi guests, example boot 4 vms
+    2) pin every guest process to one physical CPU
+    3) reboot one of the guest
+    4) check the timedrift on all the guests
+
+    :param test: QEMU test object.
+    :param params: Dictionary with test parameters.
+    :param env: Dictionary with the test environment.
+    """
+    def verify_guest_clock_source(session, expected):
+        """
+        :param session: VM session
+        :param expected: expected clocksource
+        """
+        if expected not in session.cmd(clocksource_cmd):
+            test.fail("Guest doesn't use '%s' clocksource" % expected)
+
+    ntp_cmd = params["ntp_cmd"]
+    ntp_stop_cmd = params["ntp_stop_cmd"]
+    ntp_query_cmd = params["ntp_query_cmd"]
+    clocksource = params.get("clocksource", "kvm-clock")
+    clocksource_cmd = params["clocksource_cmd"]
+    max_vms = params["max_vms"]
+    expected_time_drift = params["expected_time_drift"]
+    same_cpu = params["same_cpu"]
+
+    error_context.context("Sync host time with ntp server", logging.info)
+    status = process.system(ntp_cmd, shell=True)
+    error_context.context("Boot the guest", logging.info)
+    params["start_vm"] = "yes"
+    vm = env.get_vm(params["main_vm"])
+    env_process.preprocess_vm(test, params, env, vm.name)
+    vm.verify_alive()
+    session = vm.wait_for_login()
+
+    num = 2
+    vmid_test = random.randint(0, int(max_vms)-1)
+    sessions = [session]
+    vmnames = [params["main_vm"]]
+
+    while num <= int(max_vms):
```

Comment:
I believe all vms could be defined by config instead of creating dynamically here.

Reply:
> I believe all vms could be defined by config instead of creating dynamically here.

We can't change dynamically the number of guests via cfg file.

Reply:
> > I believe all vms could be defined by config instead of creating dynamically here.
> 
> We can't change dynamically the number of guests via cfg file.

But I see the VM count is hard-coded in your config.

Reply:
Updated, now I can simply change the number of guests even if  defining guests in cfg.